### ISSUE TITLE
bug in checkQuizCompletion

### ISSUE THREAD

**Issue by alee** [virtualcommons/port-of-mars#624]
It appears that the query logic we are using to retrieve all responses for a given user isn't retrieving actually retrieving all of the responses - this leads to people getting stuck in a tutorial loop.

For example, from the logs for user 164:

```
{"level":20,"time":1603151689746,"pid":41,"hostname":"c7b7b78b6b0e","filename":"/code/server/src/services/quiz.js","msg":"checking quiz response {\"quizResponse\":{\"id\":1051,\"questionId\":1,\"submissionId\":113,\"answer\":1,\"da
teCreated\":\"2020-10-19T23:49:35.064Z\",\"question\":{\"id\":1,\"quizId\":1,\"question\":\"Can you chat with other participants?\",\"options\":[\"No, communication is not possible in this game.\",\"Yes, in the chat box but only du
ring specially designated phases.\",\"Yes, in the chat box for the duration of the entire game.\",\"Yes, as long as System Health is greater than 35.\"],\"correctAnswer\":2,\"tutorialElementId\":\"chat\",\"order\":2}}} from user 16
4"}
{"level":20,"time":1603151689747,"pid":41,"hostname":"c7b7b78b6b0e","filename":"/code/server/src/services/quiz.js","msg":"checking quiz response {\"quizResponse\":{\"id\":1052,\"questionId\":1,\"submissionId\":113,\"answer\":2,\"da
teCreated\":\"2020-10-19T23:49:36.489Z\",\"question\":{\"id\":1,\"quizId\":1,\"question\":\"Can you chat with other participants?\",\"options\":[\"No, communication is not possible in this game.\",\"Yes, in the chat box but only du
ring specially designated phases.\",\"Yes, in the chat box for the duration of the entire game.\",\"Yes, as long as System Health is greater than 35.\"],\"correctAnswer\":2,\"tutorialElementId\":\"chat\",\"order\":2}}} from user 164"}
{"level":20,"time":1603151689747,"pid":41,"hostname":"c7b7b78b6b0e","filename":"/code/server/src/services/quiz.js","msg":"checking quiz response {\"quizResponse\":{\"id\":1053,\"questionId\":6,\"submissionId\":113,\"answer\":3,\"da
teCreated\":\"2020-10-19T23:49:56.508Z\",\"question\":{\"id\":6,\"quizId\":1,\"question\":\"At the start of each round, 25 units is subtracted from System Health. What does this represent?\",\"options\":[\"A lack of contributions f
rom the other members of your group.\",\"Solar flares striking your habitat.\",\"The costs of trade.\",\"Standard wear and tear.\"],\"correctAnswer\":3,\"tutorialElementId\":\"upkeep25\",\"order\":9}}} from user 164"}
{"level":20,"time":1603151689747,"pid":41,"hostname":"c7b7b78b6b0e","filename":"/code/server/src/services/quiz.js","msg":"checking quiz response {\"quizResponse\":{\"id\":1056,\"questionId\":3,\"submissionId\":113,\"answer\":2,\"da
teCreated\":\"2020-10-19T23:50:33.940Z\",\"question\":{\"id\":3,\"quizId\":1,\"question\":\"What happens if the System Health falls between 35 and 65?\",\"options\":[\"You can no longer chat with your group.\",\"You get an additional four time blocks.\",\"Two events will occur in this round.\",\"You receive an extra Accomplishment.\"],\"correctAnswer\":2,\"tutorialElementId\":\"upkeep65\",\"order\":4}}} from user 164"}
{"level":20,"time":1603151689747,"pid":41,"hostname":"c7b7b78b6b0e","filename":"/code/server/src/services/quiz.js","msg":"checking quiz response {\"quizResponse\":{\"id\":1059,\"questionId\":5,\"submissionId\":113,\"answer\":1,\"da
teCreated\":\"2020-10-19T23:50:40.026Z\",\"question\":{\"id\":5,\"quizId\":1,\"question\":\"If your group's System Health drops below 35 which of the following statements is true?\",\"options\":[\"The game ends.\",\"The group will
encounter three Events this round.\",\"The player with the least points is eliminated.\",\"Nothing, life as usual on Mars.\"],\"correctAnswer\":1,\"tutorialElementId\":\"upkeep35\",\"order\":8}}} from user 164"}
{"level":20,"time":1603151689748,"pid":41,"hostname":"c7b7b78b6b0e","filename":"/code/server/src/services/quiz.js","msg":"checking quiz response {\"quizResponse\":{\"id\":1076,\"questionId\":4,\"submissionId\":113,\"answer\":0,\"dateCreated\":\"2020-10-19T23:54:42.060Z\",\"question\":{\"id\":4,\"quizId\":1,\"question\":\"In each round you will have 10 time blocks to invest (unless otherwise instructed). Which of the following statements is INCORRECT?\",\"options\":[\"You can spend as many time blocks as you want.\",\"Each player has unique skills that change the costs of specific Influence Resources.\",\"Each time block invested in System Health increases your group's System Health by 1.\",\"Every time block you do not spend can be saved and used in the next round.\"],\"correctAnswer\":3,\"tutorialElementId\":\"timeblocks\",\"order\":7}}} from user 164"}
{"level":20,"time":1603151689748,"pid":41,"hostname":"c7b7b78b6b0e","filename":"/code/server/src/services/quiz.js","msg":"checking quiz response {\"quizResponse\":{\"id\":1077,\"questionId\":4,\"submissionId\":113,\"answer\":1,\"dateCreated\":\"2020-10-19T23:54:43.130Z\",\"question\":{\"id\":4,\"quizId\":1,\"question\":\"In each round you will have 10 time blocks to invest (unless otherwise instructed). Which of the following statements is INCORRECT?\",\"options\":[\"You can spend as many time blocks as you want.\",\"Each player has unique skills that change the costs of specific Influence Resources.\",\"Each time block invested in System Health increases your group's System Health by 1.\",\"Every time block you do not spend can be saved and used in the next round.\"],\"correctAnswer\":3,\"tutorialElementId\":\"timeblocks\",\"order\":7}}} from user 164"}
{"level":20,"time":1603151689748,"pid":41,"hostname":"c7b7b78b6b0e","filename":"/code/server/src/services/quiz.js","msg":"checking quiz response {\"quizResponse\":{\"id\":1078,\"questionId\":4,\"submissionId\":113,\"answer\":2,\"dateCreated\":\"2020-10-19T23:54:44.088Z\",\"question\":{\"id\":4,\"quizId\":1,\"question\":\"In each round you will have 10 time blocks to invest (unless otherwise instructed). Which of the following statements is INCORRECT?\",\"options\":[\"You can spend as many time blocks as you want.\",\"Each player has unique skills that change the costs of specific Influence Resources.\",\"Each time block invested in System Health increases your group's System Health by 1.\",\"Every time block you do not spend can be saved and used in the next round.\"],\"correctAnswer\":3,\"tutorialElementId\":\"timeblocks\",\"order\":7}}} from user 164"}
{"level":20,"time":1603151689748,"pid":41,"hostname":"c7b7b78b6b0e","filename":"/code/server/src/services/quiz.js","msg":"checking quiz response {\"quizResponse\":{\"id\":1079,\"questionId\":4,\"submissionId\":113,\"answer\":3,\"dateCreated\":\"2020-10-19T23:54:45.063Z\",\"question\":{\"id\":4,\"quizId\":1,\"question\":\"In each round you will have 10 time blocks to invest (unless otherwise instructed). Which of the following statements is INCORRECT?\",\"options\":[\"You can spend as many time blocks as you want.\",\"Each player has unique skills that change the costs of specific Influence Resources.\",\"Each time block invested in System Health increases your group's System Health by 1.\",\"Every time block you do not spend can be saved and used in the next round.\"],\"correctAnswer\":3,\"tutorialElementId\":\"timeblocks\",\"order\":7}}} from user 164"}
{"level":20,"time":1603151689748,"pid":41,"hostname":"c7b7b78b6b0e","filename":"/code/server/src/services/quiz.js","msg":"user 164 answered the following questions incorrectly: [2]"}
```

**Comment by alee** [comment]
This is still happening despite the changes in a176258bb9dbb1ceaaadefae0e4bb7f00b235535

Even the CLI tool is now reporting that users with actual quiz responses have 0 responses. Need to verify our typeorm setup to see why this is happening

**Comment by alee** [comment]
should be fixed in 389e61b795c0fb33f42851a35c520234fe0c0757 - the find options were not being interpolated correctly and the latest quiz submission was always being returned, leading to inconsistent results as lots of users used the system
